src/framework/model.js

```diff
@@ -208,4 +227,86 @@ var RaModel = GObject.registerClass({
             this.notify('hover-sound');
         }
     }
+
+    _createCSS() {
+        const scss = Gen.generateSCSS(this);
+        return Utils.transformStringToFD(scss,
+            ['sassc', '-s', '-I', '/usr/share/eos-knowledge/theme']);
+    }
+
+    _createJSON() {
+        const yaml = Gen.generateYAML(this);
+        return Utils.transformStringToFD(yaml,
+            ['autobahn', '-I', '/usr/share/eos-knowledge/preset']);
+    }
+
+    async _createGResource() {
+        const manifest = Gen.generateGResource(this);
+
+        const tmpDir = Gio.File.new_for_path(GLib.get_user_runtime_dir());
+        const logoResource = Gio.File.new_for_uri(`resource://${this._logoGraphic}`);
+        const logo = tmpDir.get_child('logo');
+        const manifestXml = tmpDir.get_child('hacking.gresource.xml');
+        const gresource = tmpDir.get_child('hacking.gresource');
+
+        const logoInputStream = await logoResource.read_async(_PRIO, null);
+        const logoOutputStream = await logo.replace_async(null, false,
+            Gio.FileCreateFlags.NONE, _PRIO, null);
+        const flags = Gio.OutputStreamSpliceFlags.CLOSE_SOURCE |
+            Gio.OutputStreamSpliceFlags.CLOSE_TARGET;
+        await logoOutputStream.splice_async(logoInputStream, flags, _PRIO, null);
+
+        await manifestXml.replace_contents_bytes_async(ByteArray.fromString(manifest),
+            null, false, Gio.FileCreateFlags.NONE, null);
+
+        const proc = new Gio.Subprocess({
+            argv: ['glib-compile-resources', '--target', gresource.get_path(),
+                '--sourcedir', tmpDir.get_path(), manifestXml.get_path()],
+        });
+        proc.init(null);
+        await proc.wait_check_async(null);
+
+        // Ah, geez, this is terrible
```

Comment:
What is this doing exactly?

Reply:
So, this is the only way I could find to open a file and get an FD using GJS ... it uses Gio.Subprocess to run the command `cat myfile`, and grabs the FD from its stdout pipe.